examples/go/filewatcher/README.md

```diff
@@ -0,0 +1,172 @@
+
+# Filewatcher
+
+A lightweight Golang CLI tool that monitors a directory for PDF file changes and sends normalized file events to an external API in real time.
+
+Designed to support downstream workflows such as AI analysis, embedding pipelines, and event logging.
+
+---
+
+## Features
+
+* Real-time directory monitoring on macOS
+* PDF-only file filtering
+* Detects file lifecycle events:
+
+  * `FILE_ADDED`
```

Comment:
Nice work outlining this

I'm curious how file renames are represented. As a `FILE_UPDATED`? or as a `FILE_REMOVED+FILE_ADDED`